### ISSUE TITLE
could not offline install Cypress

### ISSUE THREAD

**Issue by pcouas** [cypress-io/cypress#9007]
## Current behavior
Could not offline install Cypress

### Desired behavior
I want to copy cypress and offline install cypress npm package

### Test code to reproduce
A On source machine with internet, i create an `tar.gz`
```
npm install -g npm-pack-all
npm install  -g cypress@4.4.1
cd C:\Users\ZZZZ\AppData\Roaming\npm\node_modules\cypress
npx npm-pack-all
```
=> that's create tgz in cypress directory « `cypress-4.4.1.tgz` »

B on target machine without internet, tried to install `tar.gz`

After copy `tgz` to `C:\Users\zzzz\temps\XXX1\cypress\bin\npm\tgz`
```
npm install cypress-4.4.1.tgz
```

Error message is
```
 cypress@4.4.1 postinstall C:\Users\zzzz\temps\XXX1\cypress\bin\npm\tgz\node_modules\cypress
> node index.js --exec install

Installing Cypress (version: 4.4.1)

  ×  Downloading Cypress
    → Cypress Version: 4.4.1
     Unzipping Cypress
     Finishing Installation
The Cypress App could not be downloaded.

Does your workplace require a proxy to be used to access the Internet? If so, you must configure the HTTP_PROXY environment variable before downloading Cypress. Read more: https://on.cypress.io/proxy-configuration

Otherwise, please check network connectivity and try again:
```

### Versions
4.4.1

**Comment by jennifer-shehane** [comment]
Can you try following our advanced instructions for installation? Specifically the `CYPRESS_INSTALL_BINARY` env var. https://on.cypress.io/installing-cypress#Install-binary

**Comment by avestura** [comment]
According to the docs, it says specifying a file can install cypress locally instead of using the internet:
```
CYPRESS_INSTALL_BINARY=c:/cypress.zip npm install cypress
```

yet it tries to download it from the internet, even though it detects the env var:

```
Command: node index.js --exec install
Arguments: 
Directory: C:\agentT2\_work\153\s\apps\gl\node_modules\cypress
Output:
‼ Warning: Forcing a binary version different than the default.

  The CLI expected to install version: 6.3.0

  Instead we will install version: c:/cypress.zip npm install cypress

  These versions may not work properly together.

Installing Cypress (version: c:/cypress.zip npm install cypress)

5l[14:36:44]  Downloading Cypress     [started]‌
[14:37:05]  Downloading Cypress     [failed]
5hThe Cypress App could not be downloaded.‌

Does your workplace require a proxy to be used to access the Internet? If so, you must configure the HTTP_PROXY environment variable before downloading Cypress. Read more: https://on.cypress.io/proxy-configuration

Otherwise, please check network connectivity and try again:

----------

URL: https://download.cypress.io/desktop/c:/cypress.zip%20npm%20install%20cypress?platform=win32&arch=x64
Error: connect ETIMEDOUT 104.26.6.176:443

----------
```

**Comment by NicolaMLM** [comment]
I am also getting this issue with it prefixing  **h..://download.cypress.io/desktop** to the path to the tarball

This is failing in a CI environment which is behind a proxy

**Comment by jjedlicka** [comment]
This still exists in 8.3.X.  is there any progress on this fix?

**Comment by Actvist** [comment]
The command to install 8.3 without internet access does not work. Has anyone been able to do this? There should be an install option for an install without accessing the internet.

**Comment by Yolepo** [comment]
Hello,
Same problem here.
Bump!

**Comment by JessefSpecialisterren** [comment]
TL;DR double-check that the file path you're specifying actually exists! If it doesn't, you don't get a sensible error message, but a failed download of a nonsensical URL instead 😅

Ran into this issue today after installing a local zip had worked flawlessly for a month. Suddenly the postinstall started putting the file path in a download url and trying to download that. After reading the underlying code, I discovered that _Cypress will silently fall back to putting the path in a download url if the value of CYPRESS_INSTALL_BINARY does not exist as a local file_. In my case, my Jenkins workspace had been cleared somehow and the cypress zip file disappeared. You don't get an error or even a suggestion that the zip you specified cannot be found. You just get a nonsensical download request 😇

The underlying code, for those interested: https://github.com/cypress-io/cypress/blob/v9.1.1/cli/lib/tasks/install.js#L372-L423

**Comment by MaximTretjakov** [comment]
Problem still actual.
[jennifer-shehane](https://github.com/jennifer-shehane) Help!!!

**Comment by FabiEbert** [comment]
Same here

**Comment by avestura** [comment]
@MaximTretjakov one workaround for this would be to install cypress once, and then copy the `x.y.z` (cypress version) folder of cypress cache inside `C:\Users\USERNAME\AppData\Local\Cypress\Cache\x.y.z` to the same directory of the offline machine. If such a folder exists in the local cypress cache, cypress prevents to download it.

**Comment by nohuddleoff** [comment]
@avestura, I did what you suggested, copied 10.9.0 folder from the online machine and pasted it into the same directory in the offline machine.  Unfortunately, when I tried to install, it does try to fetch from registry.npmjs.org.  I got "FetchError: request to https://registry.npmjs.org/cypress failed, reason: connect ECONNREFUSED".  Of course, when I do 'ping registry.npmjs.org', I get 'request timed out'.  I tried CYPRESS_INSTALL_BINARY but failed with the same error.  Can Cypress be installed offline?

**Comment by avestura** [comment]
@nohuddleoff The method I suggested is for offline installing the cypress binary only. Of course, if you want to install other npm packages offline, including "eslint" (which you're trying to download here), you need to download and copy them offline separately. This can be done using different approaches. Either:
1. Use an offline self-hosted npm registry available on the offline machine's network. You can self-host one using Azure Artifacts, JFrog NPM repo, Sonatype Nexus repo manager, etc. or instead:
2. Just copy the `node_modules` folder from the online machine into the offline machine.

**Comment by rudishafakian** [comment]
I got around this by using Live Server extension and then giving the CYPRESS_INSTALL_BINARY a value of http://localhost:550/cypress.zip

This way it wasn't just appending to the end of the url, it was completely replacing it.  This way I could install offline.
